packages/std/src/memory.rs

```diff
@@ -50,8 +53,8 @@ pub unsafe fn consume_region(ptr: *mut c_void) -> Result<Vec<u8>, Error> {
     let region = Box::from_raw(ptr as *mut Region);
     let buffer = Vec::from_raw_parts(
         region.offset as *mut u8,
-        region.len as usize,
-        region.len as usize,
+        region.length as usize,
+        region.capacity as usize,
```

Comment:
Yes, these two look good. We just need to make sure the VM sets length properly before returning